docs/howtoguides/enable_ua_in_dockerfile.md

```diff
@@ -0,0 +1,123 @@
+# How to Enable Ubuntu Advantage Services in a Dockerfile
```

Comment:
Hello @valentincanonical please take a look at the generic how to guide here as well as the fips-centric tutorial below and let us know if you have any feedback :)

Reply:
Thanks @valentincanonical!

Totally agree on all points and will make those changes.

Some specific responses:

> why does ca-certificates have to be explicitly installed instead of being an ubuntu-advantage-tools dependency?
We're generally very conservative about adding explicit dependencies because it adds bloat to all ubuntu-minimal-based machines in existence. `ua` will actually partially function without `ca-certificates` and will tell you to install it if it's needed for a given operation. We may want to consider adding this as an actual dependency though (or perhaps a "Recommended" dep), CC @lucasmoura @renanrodrigo @CalvoM @blackboxsw.

> FROM ubuntu:bionic: does it not work with focal?
It does work with focal. I just arbitrarily chose a release for the example, but you're right that we should default to our latest LTS. I'll change it :)

Reply:
Yes, I agree with @renanrodrigo point here

Reply:
Right, because it's the same package that is used in non-container ubuntu, correct? (and the "Recommended" won't help, as we're excluding recommends.) Then that should do it!

Reply:
@orndorffgrant I tried the tutorial with the daily PPA on focal and bionic today. Works well!
Only thing I would add is: we might want to add a line at the end to advice to squash the layers (as in, they have a ubuntu-focal-fips Dockerfile to build their FIPS-enabled base image, they squash and push it to a **private** registry, and then they can use it FROM whatever else Dockerfile they need)

Reply:
@valentincanonical Glad it worked!
Just updated these docs with your suggestions - let me know what you think.

The UA team did discuss more about adding `ca-certificates` as an explicit dependency - we're going to consider adding it as a "Recommends" (which would mean removing the --no-install-recommends from this guide), but it is going to require more discussion and we shouldn't count on that happening before landing this PR.

With regards to squashing: The current tutorials are designed so that you don't need to squash. Because everything happens in one RUN command, everything is already happening in a single layer. So I'm not sure we should recommend it here. I did add a note to the how-to-guide that mentions squashing as an alternative to using one large RUN command though. Let me know if you think this is enough.